### PR TITLE
Call X.init_threads before initializing Gtk. Fix #232

### LINKED ISSUE #232: Crash when playing videos
When playing a video, `pdfpc` crashes with the following errors:
```
[xcb] Unknown request in queue while dequeuing
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
pdfpc: ../../src/xcb_io.c:179: dequeue_pending_request: Assertion `!xcb_xlib_unknown_req_in_deq' failed.
Aborted
```

This happen *only* when the presentation is run on two displays.
On a single screen, no problem occur, neither in 'presentation' mode nor in 'presenter' mode.

This is possibly a consequence of #228.

### PR DESCRIPTION
**Pull request by severin-lemaignan**
This adds a compilation dependency on X11.

### CODE REVIEW

**Comment by AndreasBilke** [comment]
But this might not work if we have a pure wayland session, mac osx or
windows?

**Comment by severin-lemaignan** [comment]
Indeed. What would be the best approach? conditional compilation to only enable that on X11 environments?

**Comment by severin-lemaignan** [comment]
(note that calling `XInitThreads` on Wayland is [not a problem](https://bugzilla.gnome.org/show_bug.cgi?id=767516#c4) 0

**Comment by AndreasBilke** [comment]
But including x11 on other systems would fail most probably. And yes,
conditional compilation might be a solution.

**Comment by severin-lemaignan** [comment]
I've updated to PR to pass a flag `X11` on non-Windows, non-Apple platforms.
I haven't checked the code compile and work on these platforms, though.

**Comment by phillipberndt** [comment]
`pkg-config --variable=targets gtk+-3.0` includes `x11` if GTK was compiled with x11 support.

**Comment by severin-lemaignan** [comment]
Well on my machine at least, the linker was complaining about missing X libs -- that's why I've explicitly added the dependency on X11.

**Comment by phillipberndt** [comment]
Sure, you need to do that. I've meant that post as a less heuristic alternative to your `NOT WIN32 AND NOT APPLE`: If GTK was compiled with X11 support, then it makes sense to include Xlib and make the `X.init_threads` call.

**Comment by AndreasBilke** [comment]
And how can you check if gtk was compiled with x11 support in cmake?

**Comment by severin-lemaignan** [comment]
@AndreasBilke @phillipberndt I'm now checking whether GTK depends on X11. That should do the trick. Still needs testing on non-X11 platforms!

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -31,6 +37,7 @@ set(CFLAGS
     ${GIO_CFLAGS} ${GIO_CFLAGS_OTHER}
     ${GEE_CFLAGS} ${GEE_CFLAGS_OTHER}
     ${POPPLER_CFLAGS} ${POPPLER_CFLAGS_OTHER}
+    ${X11_CFLAGS} ${X11_CFLAGS_OTHER}
```
Comment (AndreasBilke) [inline src/CMakeLists.txt]: Is this empty if there is no X11 package loaded?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -86,6 +95,7 @@ PACKAGES
     gio-2.0
     gee-0.8
     poppler-glib
+    x11
```
Comment (AndreasBilke) [inline src/CMakeLists.txt]: This will mostly fail on systems without X11.

**Comment by severin-lemaignan** [comment]
@AndreasBilke changes implemented

**Pull request merged**